lib/rules/no-classic-classes.js

```diff
@@ -50,7 +50,10 @@ module.exports = {
           const classImportedFrom = getSourceModuleNameForIdentifier(context, node.object);
 
           // Only warn about classes imported from the `@ember/` namespace
-          if (classImportedFrom && classImportedFrom.startsWith('@ember/')) {
+          if (
+            classImportedFrom &&
+            (classImportedFrom.startsWith('@ember/') || classImportedFrom.includes('ember-data'))
```

Comment:
`classImportedFrom.includes('ember-data')` may be a bit loose, the alternative is to do a `startsWith` `@ember-data` and `ember-data` separately.

Thoughts?

Reply:
Actually thinking about it, should this be made configurable? 🤔

We could support `@ember/` `@ember-data/` and `ember-data` out-of-the-box, but then you can pass your own set of strings to check with `String#startsWith`. This way I can make sure to use native classes for the in-built classes but also subclassed components/classes or other libraries in the addon ecosystem?

Reply:
Configurability could be added later in a separate PR if desired.

For now, let's split this apart into the two checks (starts with `@ember-data/` or exactly equals `ember-data`).